resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientInvocation.java

```diff
@@ -191,7 +208,19 @@ public ClientInvocation(final ResteasyClient client, final URI uri, final Client
             }
          }
          if (status >= 300 && status < 400)
-            throw new RedirectionException(response);
+         {
+            Config config = ConfigProvider.getConfig();
+            boolean originalBehavior = config.getOptionalValue(ResteasyContextParameters.RESTEASY_ORIGINAL_WEBAPPLICATIONEXCEPTION_BEHAVIOR, boolean.class).orElse(false);

```

Comment:
Can this be moved to a private method to be invoked from the "if" below (in order to avoid duplication below and benefit from the short-circuit behaviour of the || operator)?

Reply:
Nice. Done.